File: src/main/java/br/com/useinet/finance/dto/CategoriaResponse.java
package br.com.useinet.finance.dto;

import br.com.useinet.finance.model.Categoria;

public class CategoriaResponse {
    private Long id;
    private String nome;

    public static CategoriaResponse from(Categoria c) {
        CategoriaResponse r = new CategoriaResponse();
        r.id = c.getId();
        r.nome = c.getNome();
        return r;
    }

    public Long getId() { return id; }
    public String getNome() { return nome; }
}

File: src/main/kotlin/br/com/useinet/finance/dto/CategoriaResponse.kt
package br.com.useinet.finance.dto

import br.com.useinet.finance.model.Categoria

data class CategoriaResponse(val id: Long, val nome: String) {
    companion object {
        fun from(c: Categoria) = CategoriaResponse(
            id = requireNotNull(c.id) { "Categoria.id não pode ser nulo" },
            nome = requireNotNull(c.nome) { "Categoria.nome não pode ser nulo" }
        )
    }
}
